pin/pin.go

```diff
@@ -221,30 +231,51 @@ func (p *pinner) RemovePinWithMode(key util.Key, mode PinMode) {
 func LoadPinner(d ds.ThreadSafeDatastore, dserv mdag.DAGService) (Pinner, error) {
 	p := new(pinner)
 
+	rootKeyI, err := d.Get(pinDatastoreKey)
```

Comment:
I think the key needs to be a valid object key? (i.e. be a hash) 

i think right now it's just at a filename named `"$IPFS_PATH/flatfs/local/pins"`. which is how git does branches / tags. but git tags/branches are not objects themselves (though perhaps they should be). so in our case, we can make an object for this, and store it at `"$REPO_ROOT/pins"` where `$REPO_ROOT` is the hash of an object too; an object representing local repo state. (lmk if this isn't clear)